docs/classic-docs/docs/capturing/index.md

```diff
@@ -325,3 +325,25 @@ Tracee can capture the following types of artifacts:
         Example kernel module taken from [this blog]
 
 [this blog]: https://blog.sourcerer.io/writing-a-simple-linux-kernel-module-d9dc3762c234
+
+1. **BPF programs**
+
+    Anytime a **BPF program** is loaded, the BPF bytecode will be captured.
+
+     ```text
+     $ sudo ./dist/tracee-ebpf \
+        --output none \
+        --filter comm=bash \
+        --filter follow \
+        --capture clear-dir \
+        --capture bpf
+     ```
+
+    Captured module will be found in tracee destination directory, just like
+    any other captured file would be:
+
+     ```text
+     $ sudo ls /tmp/tracee/out/host
+       bpf.name-test_prog.pid-3668786.c8b62228208f4bdbf21df09c01046b73dd44733841675bf3c0ff969fbedab616
```

Comment:
What is this file format ? Is it an ELF without the ELF header ? How can I use, read and/or disasemble it ? What would it be good for ? Note that the kernel module (from the above capture) is loadable, and a proper object file. This file is just "data" for me when loading a very simple eBPF object.

Reply:
Alright, so... if you install `binutils-bpf` in `ubuntu`, you're able to disassemble the raw binary by setting the specific arch. There is also a way of renaming/creating the ELF sections with llvm-objcopy but I wasn't able to.

This is the bytecode I got from the object that was dumped:

```
# /usr/lib/bpf/bin/objdump -D -b binary -m bpf ./bpf.name-kprobe__sys_mma.pid-86815.3c670bbf686f
fa2bb1fa1e2f2b2c21175ffa4bfd612703f5f7c4dfbc73c61d87

./bpf.name-kprobe__sys_mma.pid-86815.3c670bbf686ffa2bb1fa1e2f2b2c21175ffa4bfd612703f5f7c4dfbc73c61d87:     file format binary


Disassembly of section .data:

0000000000000000 <.data>:
   0:   b7 02 00 00 e5 07 00 00         mov %r2,0x7e5
   8:   63 2a fc ff 00 00 00 00         stxw [%fp+-4],%r2
  10:   bf a4 00 00 00 00 00 00         mov %r4,%fp
  18:   07 04 00 00 fc ff ff ff         add %r4,-4
  20:   18 12 00 00 08 00 00 00         lddw %r2,8
  28:   00 00 00 00 00 00 00 00
  30:   18 03 00 00 ff ff ff ff         lddw %r3,0xffffffff
  38:   00 00 00 00 00 00 00 00
  40:   b7 05 00 00 04 00 00 00         mov %r5,4
  48:   85 00 00 00 19 00 00 00         call 25
  50:   b7 00 00 00 00 00 00 00         mov %r0,0
  58:   95 00 00 00 00 00 00 00         exit
```

This is the bytecode from the binary that was loaded:

```
$ /usr/lib/bpf/bin/objdump -j kprobe/sys_mmap -D -m bpf ./main.bpf.o

./main.bpf.o:     file format elf64-bpfle

Disassembly of section kprobe/sys_mmap:

0000000000000000 <kprobe__sys_mmap>:
   0:   b7 02 00 00 e5 07 00 00         mov %r2,0x7e5
   8:   63 2a fc ff 00 00 00 00         stxw [%fp+-4],%r2
  10:   bf a4 00 00 00 00 00 00         mov %r4,%fp
  18:   07 04 00 00 fc ff ff ff         add %r4,-4
  20:   18 02 00 00 00 00 00 00         lddw %r2,0
  28:   00 00 00 00 00 00 00 00
  30:   18 03 00 00 ff ff ff ff         lddw %r3,0xffffffff
  38:   00 00 00 00 00 00 00 00
  40:   b7 05 00 00 04 00 00 00         mov %r5,4
  48:   85 00 00 00 19 00 00 00         call 25
  50:   b7 00 00 00 00 00 00 00         mov %r0,0
  58:   95 00 00 00 00 00 00 00         exit
```

confirming the raw eBPF bytecode saved from the code is the same.